modules/programs/vscode.nix

```diff
@@ -0,0 +1,31 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.vscode;
+
+in
+
+{
+  options = {
+    programs.vscode = {
+      settings = mkOption {
+        type = types.attrs;
+        default = {};
+        description = ''
+          Configuration written to
+          <filename>~/.config/Code/User/settings.json</filename>
+        '';
+      };
+    };
+  };
+
+  config = mkIf (cfg.settings != {}) {
+    xdg.configFile."Code/User/settings.json".text = ''
+      // Generated by Home Manager.
```

Comment:
Since this is a json file I suppose it is best to leave out any comments. So
```nix
xdg.configFile."Code/User/settings.json".text = builtins.toJSON cfg.settings;
```
should be enough.